toolkit/tools/imagecustomizerapi/iso.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerapi
+
+import (
+	"fmt"
+)
+
+// Iso defines how the generated iso media should be configured.
+type Iso struct {
+	AdditionalFiles map[string]FileConfigList `yaml:"AdditionalFiles"`
+}
+
+func (s *Iso) IsValid() error {
+	var err error
+
+	for sourcePath, fileConfigList := range s.AdditionalFiles {
+		err = fileConfigList.IsValid()
+		if err != nil {
+			return fmt.Errorf("invalid file configs for (%s):\n%w", sourcePath, err)
+		}
+	}
```

Comment:
Instead of erroring out for every invalid file config, it may be better to accumulate the errors in an array and return a single error from the function, if the array length is > 0, which would contain all the error messages for all invalid configs.

Reply:
fixed.